### ISSUE TITLE
Selfhosted GitLab CE in subdirectory not working

### ISSUE THREAD

**Issue by jancbeck** [afragen/github-updater#302]
I have a my selfhosted GitLab CE installation in a subdirectory e.g. https://mydomain.com/gitlab/
This is possible by setting `relative_url_root ` in `gitlab/gitlab/config/gitlab.yml` to the relative path (`/gitlab` in my case).

This changes my repo URLs to something like this
**`https://mydomain.com/gitlab/jancbeck/my-project/`**
(`https://mydomain.com/gitlab/$namespace/$repo/`)

However, the installer seems to assume that all GitLab installs run from the root directory of the domain and strips away the path of the base uri:
https://github.com/afragen/github-updater/blob/78264a8e80f09ada95bb64cece559ff281be5ac0/src/GitHub_Updater/Base.php#L565

Thus changing my download URLs to something like:
`https://mydomain.com/gitlab/jancbeck/repository/archive.zip?ref=master&private_token=...`

Which of course returns:
> **Download failed. Not Found.**

**Comment by afragen** [comment]
Yes, that is a limitation of how support for GitLab CE, GitLab Enterprise or GitHub Enterprise is enabled. Unfortunately, parsing the URI is much simpler for the domain. Having an installation in a subdirectory, like subdirectory locations for plugin or theme repos, doesn't work.

You might have to create a redirect to pass `http://gitlab.example.com` to `http://example.com/gitlab/`

**Comment by jancbeck** [comment]
Thanks, I created the redirect and that allowed me to install a plugin via the installer. However, updating doesn't seem to work. Should I put my redirect URL into the plugin header tags `GitLab CE` / `GitLab Plugin URI` as well?

**Comment by afragen** [comment]
You might have to reinstall GitLab into a subdomain for it to work with this plugin. That or some other solution, sorry.

**Comment by jancbeck** [comment]
I tried the subdomain solution via an .htaccess rule:

```
RewriteEngine On
RewriteCond %{HTTP_HOST} =git.mydomain.com
RewriteRule (.*) https://mydomain.com/gitlab/$1 [R=301,L]
```

This correctly resolves the plugin URIs to their gitlab package when installing. 

The problem with updates come from this check in https://github.com/afragen/github-updater/blob/develop/src/GitHub_Updater/GitLab_API.php#L420

It prevented the updates from working because I have named my project differently than the theme slug. I changed the line to

```
if ( $this->type->repo === $project->path ) {
```

That worked, but I guess I could also rename the project on Gitlab to match the slug.

The [readme](https://github.com/afragen/github-updater#themes) is a bit misleading here:
> When initially adding a theme, the directory must be identical to the repo name.

I had no problem initially adding the theme, but updates require an identical name.
